src/sentry/replays/consumers/recording/process_recording.py

```diff
@@ -0,0 +1,191 @@
+        recording_file_name = (
+            f"rr:{message_dict['replay_id']}:{message_dict['recording_headers']['sequence_id']}"
+        )
+
+        file = File.objects.create(
+            name=recording_file_name,
+            type="replay.recording",
+        )
+        file.putfile(
+            BytesIO(recording),
+            blob_size=settings.SENTRY_ATTACHMENT_BLOB_SIZE,
+        )
+        # associate this file with an indexable replay_id via ReplayRecordingSegment
+        ReplayRecordingSegment.objects.create(
+            replay_id=message_dict["replay_id"],
+            project_id=message_dict["project_id"],
+            sequence_id=message_dict["recording_headers"]["sequence_id"],
+            file_id=file.id,
+        )
+
+    def _get_from_cache(self, message_dict):
+        replay_recording = message_dict["replay_recording"]
+        id = message_dict["replay_recording"]["id"]
+        project_id = message_dict["project_id"]
+        cache_id = replay_cache_id(id, project_id)
+        replay_payload = attachment_cache.get_from_chunks(key=cache_id, **replay_recording)
+        try:
+            replay_data = replay_payload.data
+        except MissingAttachmentChunks:
+            logger.warning("missing replay recording chunks!")
+            return None
+        replay_payload.delete()
+        return replay_data
+
+    def _process_recording(
+        self, message_dict: RecordingMessage, message: Message[KafkaPayload]
+    ) -> None:
+        recording = self._get_from_cache(message_dict)
+        # split the recording payload by a newline into the headers and the recording
+        try:
+            recording_headers, recording = recording.split(b"\n", 1)
+        except ValueError:
+            logger.warning(f"no headers on recording {message_dict['replay_id']}, dropping")
+            return
+        message_dict["recording_headers"] = json.loads(recording_headers)
+
+        # Upload the combined to the datastore.
+        self.__futures.append(
+            ReplayRecordingMessageFuture(
+                message,
+                self.__threadpool.submit(
+                    self._store, message_dict=message_dict, recording=recording
+                ),
+            )
+        )
+
+    def submit(self, message: Message[KafkaPayload]) -> None:
+        assert not self.__closed
+
+        # TODO: validate schema against json schema?
+        try:
+            message_dict = msgpack.unpackb(message.payload.value)
+
+            if message_dict["type"] == "replay_recording_chunk":
+                self._process_chunk(cast(RecordingChunkMessage, message_dict), message)
+            if message_dict["type"] == "replay_recording":
+                self._process_recording(cast(RecordingMessage, message_dict), message)
+        except Exception as e:
+            # avoid crash looping on bad messsages for now
+            sentry_sdk.capture_exception(e)
+            logger.error(e)
```

Comment:
You're better off just using `logger.exception("Failed to process message")`. The exception info will be sent automatically